### DISCUSSION TITLE
All next/image components failing load default image and handling error,  if image doesn't exist ( HTTP ERROR 404 for image ) on remote path.

### DISCUSSION THREAD

**Discussion by ashish-dappgenie** [vercel/next.js#69067] [Help]
### Summary

I am using next js 14.2.5 and facing an issue in the next js Image component, in an array of data whenever there is a remote image link that throws 404 I want to load the default image at that time. I am doing so by using the onError event listener but that is also not working. Please provide a way around to solve this problem.  

`onError={(e) => {
                e.currentTarget.src = defaultTeamLogo;
              }}
              `

**Comment by 3liBawazeer** [comment]
**Use This Comonent**

`
'use client';

import { useEffect, useState } from 'react';
import Image, { ImageProps } from 'next/image';

// Define the component's props with some restrictions on `ImageProps`
interface SafeImageProps extends Omit<ImageProps, 'src' | 'width' | 'height'> {
  src: string; // Primary image source
  fallbackSrc?: string; // Default image source (optional)
  alt: string; // Alt text for accessibility
  width?: number; // Optional width for the image
  height?: number; // Optional height for the image
}

const SafeImage: React.FC<SafeImageProps> = ({
  src, // Primary image source
  fallbackSrc = '/images/sedebadLogo.svg', // Default fallback image
  alt = 'Image', // Default alt text
  width,
  height,
  ...rest // Spread other props like className or priority
}) => {
  const [validSrc, setValidSrc] = useState<string>(fallbackSrc); // State to track the valid image source

  // Run validation whenever the `src` changes
  useEffect(() => {
    checkImage(src); // Check and set the appropriate image source
  }, [src]);

  /**
   * Validates the provided `src` and sets `validSrc`.
   * - Adds a leading slash for relative paths.
   * - Accepts HTTPS/HTTP links as-is.
   */
  const checkImage = (src: string) => {
    if (!src.startsWith('https') && !src.startsWith('http')) {
      setValidSrc(src.startsWith('/') ? src : '/' + src); // Ensure valid relative path
      return;
    }
    setValidSrc(src); // Use the absolute URL as-is
  };

  return (
    <Image
      src={validSrc} // The current valid source
      loader={() => validSrc} // Custom loader for manual source
      alt={alt} // Alt text for the image
      onError={() => setValidSrc(fallbackSrc)} // Handle errors and fallback to default image
      width={width} // Pass width if provided
      height={height} // Pass height if provided
      {...rest} // Spread additional props like className or priority
    />
  );
};

export default SafeImage;

`
